### ISSUE TITLE
Thermostat's logic is fussy at best

### ISSUE THREAD

**Issue by thingray** [mill1000/midea-ac-py#123]
'id': 151732605149392
'online': True
'supported': True
'type': <DeviceType.AIR_CONDITIONER:   172>

 Don't think it's an integration issue, but want to report it to document it.
48000 bTU split system on central unit. make  CANAIR, but actually a Midea.  Has the wifi  option installed.
phone app NetHome.

what it looks like  is whenever the temp has been stable for awhile, the thermostat goes to sleep.

waking it  up ( changing set point)  with the Tstat seems ok
same with nethome
But doing it with HA integration here what I observe:
starting conditions   set point 22  measured 22  ( in Celcius)  on all the devices
HA set point changed to 21
HA measured goes to 24.
Nethome set point goes to 21
Nethome measured goes to 24 
Tstat set point goes to 21
Tstat measured stays to 22.

Independant temp measurement at the Tstats confirm 22 stable

This is weird.  it also seems to trigger a fan/heat cycle for a short time.  The fact that HA behaves like  Nethome tells me it is not the integration.

I think the wifi kit on the unit has some serious issues. And I am not super impressed by the thermostat's features, Unable to have the fan cycle and the heat kick in if temp drop.  I am forced to switch mode  to heat manually.    Again not the integration at fault IMO.

Insights appreciated, and thanks for all the work,
HA  docker version 2024.3.3  Midea smartAc 2024.4.0

**Comment by v1k70rk4** [comment]
Hello, I had such strange things happen when my wifi module was the SK103X. Then I bought another air conditioner with SK105 in it and had no issues with it. After that, I replaced the modules in all my air conditioners with SK105.

What kind do you have in yours?

**Comment by v1k70rk4** [comment]
By the way, I'll write it down here too :) I only use the built-in thermostat of the air conditioner as a correction factor. I installed an external thermometer at a proper distance from the air conditioner and a helper entity to set the desired temperature.

I calculate the difference between the thermostat and the actual temperature and adjust the set point temperature so that the air conditioner will definitely cool if needed :) It has been working perfectly for 2 years.

I have one air conditioner that gets direct sunlight on it; it's not uncommon to have a 3-4 degree difference there, but even in completely shaded locations, there are such variations:
Hőmérséklet eltérés - Klíma-Szoba (Temperature Difference - Air Conditioner-Room)
![image](https://github.com/mill1000/midea-ac-py/assets/120401244/e751f15c-a3f7-49d8-bc49-494877d0b4b7)

**Comment by mill1000** [comment]
> waking it up ( changing set point) with the Tstat seems ok
> same with nethome
> But doing it with HA integration here what I observe:
> starting conditions set point 22 measured 22 ( in Celcius) on all the devices
> HA set point changed to 21
> HA measured goes to 24.
> Nethome set point goes to 21
> Nethome measured goes to 24
> Tstat set point goes to 21
> Tstat measured stays to 22.

Might be worth collecting logs with debugging turned on.

**Comment by thingray** [comment]
Tucker for the logs,  I will at some point. just no time right now.
BTW, you can add CANAIR/DESCAIR as compatible central systems in your list.  Canair is a local canadian distributor rebadging Midea.
My first step before the logs will be to go back at them and have them explain to me why I can't get  the fan circulation the air and the cooling or heating kicking in as required when needed.  This is basic thermostat function   but not to midea it seems.

 RViktor   I have the US-OSK103 - If you have any trail of documented issues with  the 103  That will help, if not I will digg myself.  I am pretty sure their local support will be a bit unaware of anything.

**Comment by v1k70rk4** [comment]
@thingray Unfortunately, I didn't document it, but the problem was clear because I had three identical air conditioners, only with a 1-year difference between each.

When I started using Home Assistant, the one with the EU-SK103X module would consistently freeze if I sent too many commands. I replaced it with the EU-SK105 from another device, and it worked perfectly, but the other one started freezing and stopped accepting commands. Since module only costs 20 euros, I didn't bother with the warranty (especially since it worked fine through the app); I just bought another module. Because of the thermostat, it's not uncommon for it to receive frequent temperature adjustments from HA.

There are many air conditioners around me, and I can monitor their functioning. Those without an external thermometer estimate the temperature up high and right before the cooling/heating medium. My previous Daikin model could be off by up to two degrees :) I like Midea because it allows local operation without the internet, and I can control it well with rules. I can't achieve such good results with my parents' Mitsubishi or my old Daikin air conditioner. Now, every room is exactly at the temperature set in HA, with T±0.2°C.
